src/ape_ethereum/multicall/handlers.py

```diff
@@ -0,0 +1,288 @@
+                    call.encode_input(*args),
+                )
+            )
+
+        elif self._handler_method_abi == AGGREGATE3VALUE_METHOD:
+            self.calls.append(
+                (
+                    call.contract.address,
+                    requireSuccess if requireSuccess is not None else True,
+                    value or 0,
+                    call.encode_input(*args),
+                )
+            )
+
+
+class Call(BaseMulticall):
+    """
+    Create a sequence of calls to execute at once using ``eth_call`` via the Multicall3 contract.
+
+    Usage example::
+
+        from ape_ethereum import multicall
+
+        call = multicall.Call()
+        call.add(contract.myMethod, *call_args)
+        call.add(contract.myMethod, *call_args)
+        ...  # Add as many calls as desired
+        call.add(contract.myMethod, *call_args)
+        a, b, ..., z = call()  # Performs multicall
+    """
+
+    def __init__(self) -> None:
+        super().__init__(ContractCallHandler)
+
+        self.abis: List[MethodABI] = []
+        self._result: Union[None, Tuple[int, List[HexBytes]], List[Tuple[bool, HexBytes]]] = None
+
+    def add(self, call: ContractMethodHandler, *args, **kwargs):
+        if "value" in kwargs:
+            raise InvalidOption("value")
+
+        super().add(call, *args, **kwargs)
+        self.abis.append(_select_method_abi(call.abis, args))
+
+    @property
+    def returnData(self) -> List[HexBytes]:
+        if not self._result:
+            raise NotExecutedError()
+
+        elif isinstance(self._result, tuple):
+            # Call3[] or Call3Value[]
+            return list(r[1] for r in self._result)  # type: ignore[index]
+
+        else:
+            # blockNumber: uint256, returnData: Call[]
+            return self._result.returnData  # type: ignore[attr-defined]
+
+    def _decode_results(self) -> Iterator[Any]:
+        for abi, data in zip(self.abis, self.returnData):
+            result = self.provider.network.ecosystem.decode_returndata(abi, data)
+
+            if isinstance(result, (list, tuple)) and len(result) == 1:
+                yield result[0]
+
+            else:
+                yield result
+
+    def __call__(self, **call_kwargs) -> Iterator[Any]:
+        """
+        Perform the Multicall call. This call will trigger again every time the ``Call`` object
+        is called.
+
+        Raises:
+            :class:`~ape_ethereum.multicall.exceptions.UnsupportedChain`:
+              If there is not an instance of Multicall3 deployed
+              on the current chain at the expected address.
+
+        Args:
+            **call_kwargs: the kwargs to pass through to the call handler.
+
+        Returns:
+            Iterator[Any]: the sequence of values produced by performing each call stored
+              by this instance.
+        """
+        self._result = self.handler(self.calls, **call_kwargs)  # type: ignore [operator]
```

Comment:
this space causes issues

```suggestion
        self._result = self.handler(self.calls, **call_kwargs)  # type: ignore[operator]
```

but would love to mypy working better in this module, there are many type ignores.

Reply:
yeah I agree

Reply:
I _think_ #1230 might actually make this a lot better, since I can just use a dict with different members, and it will only `getattr` ones that matter when making the appropiate calls